source/WindowScroller/WindowScroller.js

```diff
@@ -138,25 +136,24 @@ export default class WindowScroller extends React.PureComponent<Props, State> {
     this._isMounted = true;
   }
 
-  componentWillReceiveProps(nextProps: Props) {
-    const scrollElement = this.props.scrollElement;
-    const nextScrollElement = nextProps.scrollElement;
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    const {scrollElement} = this.props;
 
     if (
-      scrollElement !== nextScrollElement &&
-      scrollElement &&
-      nextScrollElement
+      prevProps.scrollElement !== scrollElement &&
+      prevProps.scrollElement != null &&
+      scrollElement != null
     ) {
-      this.updatePosition(nextScrollElement, nextProps);
+      this.updatePosition(scrollElement);
 
-      unregisterScrollListener(this, scrollElement);
-      registerScrollListener(this, nextScrollElement);
+      // $FlowFixMe Why is this necessary? There's a check right above.
+      unregisterScrollListener(this, prevProps.scrollElement);
+      registerScrollListener(this, scrollElement);
 
-      this._unregisterResizeListener(scrollElement);
-      this._registerResizeListener(nextScrollElement);
+      this._unregisterResizeListener(prevProps.scrollElement);
+      this._registerResizeListener(scrollElement);
     }
   }
-
```

Comment:
why remove newline?

Reply:
Probably an accident. This is a WIP PR. Not quite ready for nit-review yet 😉

Reply:
(I'll do a final cleanup pass myself once I'm done with the major refactor)